Python/pywarpx/fields.py

```diff
@@ -528,9 +531,9 @@ def __setitem__(self, index, value):
                 mf_arr = self._get_field(mfi)
                 if isinstance(value, np.ndarray):
                     slice_value = value3d[global_slices]
-                    if cp is not None:
+                    if libwarpx.libwarpx_so.Config.have_gpu:
                         # Copy data from host to device
-                        slice_value = cp.asarray(value3d[global_slices])
+                        slice_value = slice_value.to_numpy(copy=True)
```

Comment:
Careful, does that work? `.to_numpy()` is only implemented in pyAMReX types. After line 
```py
slice_value = value3d[global_slices]
```
above, what do we have as type in `slice_value`?